chatsky/core/ctx_dict.py

```diff
@@ -0,0 +1,274 @@
+from __future__ import annotations
+from asyncio import gather
+from hashlib import sha256
+import logging
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    Generic,
+    List,
+    Mapping,
+    Optional,
+    Sequence,
+    Set,
+    Tuple,
+    Type,
+    TypeVar,
+    Union,
+    overload,
+    TYPE_CHECKING,
+)
+
+from pydantic import BaseModel, PrivateAttr, TypeAdapter, model_serializer, model_validator
+
+from chatsky.utils.logging import collapse_num_list
+
+if TYPE_CHECKING:
+    from chatsky.context_storages.database import DBContextStorage
+
+K = TypeVar("K", bound=int)
+V = TypeVar("V")
+
+logger = logging.getLogger(__name__)
+
+
+def get_hash(string: bytes) -> bytes:
+    return sha256(string).digest()
+
+
+class ContextDict(BaseModel, Generic[K, V]):
+    _items: Dict[K, V] = PrivateAttr(default_factory=dict)
+    _hashes: Dict[K, int] = PrivateAttr(default_factory=dict)
+    _keys: Set[K] = PrivateAttr(default_factory=set)
+    _added: Set[K] = PrivateAttr(default_factory=set)
+    _removed: Set[K] = PrivateAttr(default_factory=set)
+
+    _storage: Optional[DBContextStorage] = PrivateAttr(None)
+    _ctx_id: str = PrivateAttr(default_factory=str)
+    _field_name: str = PrivateAttr(default_factory=str)
+    _value_type: Optional[TypeAdapter[Type[V]]] = PrivateAttr(None)
+
+    @classmethod
+    async def new(cls, storage: DBContextStorage, id: str, field: str, value_type: Type[V]) -> "ContextDict":
```

Comment:
My issues with the new creation methods stem from the fact that simply initializing context with `Context()` will no longer produce a functional context but you would also need to set `_value_type` for all context dicts:
https://github.com/deeppavlov/chatsky/blob/46e0112ff9ff98306cfcaaea35d5592144753707/tests/conftest.py#L89-L92
Which is not convenient for testing and debugging.

I think the best solution is to add validators to `Context` that would prep its context dicts (by setting their value types) so that `Context()` is a functional context without db connection.